src/main/java/org/marketdesignresearch/mechlib/core/bid/bundle/BundleExactValuePair.java

```diff
@@ -65,13 +65,17 @@ public BundleExactValuePair(BigDecimal amount, Set<Good> bundle, String id) {
     public BundleExactValuePair reducedBy(BigDecimal amount) {
         return new BundleExactValuePair(getAmount().subtract(amount).max(BigDecimal.ZERO), bundle, id);
     }
+    
+    public BundleExactValuePair multiply(BigDecimal amount) {
+        return new BundleExactValuePair(getAmount().multiply(amount), bundle, id);
+    }
 
     public BundleExactValuePair withAmount(BigDecimal amount) {
         return new BundleExactValuePair(amount, bundle, id);
     }
 
     public PotentialCoalition getPotentialCoalition(Bidder bidder) {
-        return new PotentialCoalition(getGoods(), bidder, amount);
+        return new PotentialCoalition(getBundle(), bidder, amount);
     }
```

Comment:
I don't have a deep understanding of the CCG implementation. To have it work with "generic" MRVM I had to change this here. Is this correct?

Reply:
Good catch, it would otherwise transform the bundle into a bundle of single goods. Do you by any chance have some tests available to test Benedikt's code at least a simple "generic" example

Reply:
I added a two tests for a simple example that uses two identical goods, written as one good with availability two. It seems like the default CCG that SATS is using (quadratic vcg-nearest), it behaves as expected. When other configurations are used, it results in different results. We should have a close look into this to make Benedikt's CCG code completely compatible with generic worlds. Good news is that for now, the CCG version used in SATS seems to be functional. Maybe it would be an option to not allow generic worlds for other CCG Constraint Generators than "StandardCCGConstraintGenerator" in the first version before this is fixed, to avoid wrong results... What do you think?

Reply:
I think we should have a deeper look into this issue. Also for the old sats version ccg sometimes fails with an error. When I remember correctly this is also a MIP infeasible exception.

Reply:
Can you confirm that it fails even with the "StandardCCGConstraintGenerator"? The old SATS version used a very light version of CCG, which is not equivalent with the more sophisticated StandardCCGConstraintGenerator of Benedikt.

Reply:
These are actually two different things. The StandardCCGConstraintGenerator should work with generic worlds. See also the new conversation.